hack/fuse-demo/wrap_datamon.sh

```diff
@@ -20,113 +31,236 @@ POLL_INTERVAL=1 # sec
 #####
 
 typeset COORD_POINT
-typeset -a DATAMON_CMDS
-typeset BUNDLE_ID_FILE
 
 typeset CONTEXT_BUCKET_NAME
 typeset CONTEXT_NAME
 
 SLEEP_INSTEAD_OF_EXIT=
 
-while getopts b:a:sc:d:i: opt; do
-    case $opt in
-        (s)
-            SLEEP_INSTEAD_OF_EXIT=true
-            ;;
-        (c)
-            COORD_POINT="$OPTARG"
-            ;;
-        (d)
-            datamon_cmd=$(echo "$OPTARG" |tr '\t' ' ' |tr -s ' ' | \
-                              sed 's/^ //' |sed 's/ $//')
-            DATAMON_CMDS=($datamon_cmd $DATAMON_CMDS)
-            ;;
-        (i)
-            BUNDLE_ID_FILE="$OPTARG"
-            ;;
-        (b)
-            CONTEXT_BUCKET_NAME="$OPTARG"
-            ;;
-        (a)
-            CONTEXT_NAME="$OPTARG"
-            ;;
-        (\?)
-            echo "Bad option, aborting."
-            return 1
-            ;;
-    esac
-done
-if [[ "$OPTIND" -gt 1 ]]; then
-    shift $(( OPTIND - 1 ))
+####
+# bridge parameters to shell script specific format
+# (currently environment variables
+
+if [[ -n $dm_fuse_params ]]; then
+    typeset dm_fuse_params_val
+    env_vars_file=/tmp/env_vars_from_params.sh
+    if [[ -f $dm_fuse_params ]]; then
+        dbg_print "parsing parameters YAML file $dm_fuse_params"
+        dm_fuse_params_val=$(cat $dm_fuse_params)
+    else
+        dbg_print "parsing parameters YAML from env var"
+        dm_fuse_params_val=$dm_fuse_params
+    fi
+    print -- $dm_fuse_params_val | \
+        datamon_sidecar_param parse fuse > $env_vars_file
```

Comment:
If I understand correctly, you've implemented a new datamon command-line endpoint, invoked here, that deserializes the yaml into a set of env vars that are saved in the env vars file. Is that correct?

Reply:
yes, although this is an implementation detail that's intended to be opaque from your review